index.html

```diff
@@ -3951,41 +3951,191 @@ <h2>Syntaxes</h2>
         <h3>JSON-LD</h3>
 
         <p>
+The world of credentials is vast and varied. A physical wallet places
+relatively few constraints on the shape and size of those credentials–and it
+places no restriction on their contents. Likewise, Verifiable Credential Data
+Model’s use of [[!JSON-LD]] provides the consistent shape for all compliant
```

Comment:
I'd argue JSON Schema provides a more consistent "shape" than LD does...

Reply:
> I'd argue JSON Schema provides a more consistent "shape" than LD does...

JSON schema doesn't provide a more consistent shape; it is an enforcement or validation mechanism for whatever idiosyncratic shape the author wants to use. There's no implicit constraint in how the author does their modeling, rather, JSON schema is a mechanism to enable testing whether some particular data fits their chosen shape (which can be anything). That's certainly useful and complementary technology, but not the same thing being said here.

The text in this PR is about the advantage of consistent data modeling through simple constraints such as using an object-oriented approach and the fact that this shape consistently maps to atomic statements (claims) of subject => predicate => value.

Reply:
@decentralgabe JSON Schema (by design) only validates a "surface syntax" not any underlying data modeling. JSON-LD provided mapping tree structures into a graph-based data model that can exist across many different tree "shapes."

There can't ever be a single JSON Schema to validate all JSON structures, but there can be "universal" JSON-LD storage, construction, re-construction, aggregation, etc. because the data model remains the same regardless of the tree shape.

I realize the use of the word "shape" in the physical wallet analogy causes that confusion...I'll look into how we could resolve that, but I know of no other way to provide a universally available data structure in JSON (or even across CBOR, YAML, etc.) that can survive the surface syntax shift other than JSON-LD.

Reply:
@dlongley my point is that JSON LD usage commonly defines a large set of terms and then in practice a subset of those terms are used without any way for the context creator to dictate which terms are required or not -- that's not really a shape. A JSON Schema on the other hand allows specifying required and optional properties, which are constraints I view was necessary to even have a 'shape.'

@BigBlueHat you're correct re:surface syntax vs 'underlying' data modeling if you consider semantics and graphs to be a requisite for data modeling.

I'd recommend rephrasing to emphasize semantics, graph modeling, and term definition space instead of 'shape.'

Reply:
@decentralgabe,

I think you actually want both (JSON-LD and JSON schema) -- and they apply different kinds of constraints on the "shape" of data.

JSON-LD constrains the modeling and how it will be interpreted consistently as described above, and JSON schema enforces whatever schema someone invents to describe that modeling. JSON schema alone doesn't constrain what gets modeled or how it will be interpreted; you can make whatever schema you want and JSON schema says nothing about the interpretation -- but it will enforce syntactic expression. JSON-LD alone doesn't add as much enforcement of its syntactic constraints as JSON schema does, but it ensures the modeling will be interpreted according to the constraints even if the author tries to deviate. So both are useful tools that play different roles in shaping data (model wise and syntactically). Neither has a single claim to "shape" constraining, they just work in different dimensions.